tests/test_views.py

```diff
@@ -210,16 +213,129 @@ def setUp(self):
+    def test_course_waffle_inactive(self, mock_waffle):
+        mock_waffle.return_value = False
+        message_count = 5
+        response = self.client.get(
+            reverse('message-history', kwargs={'course_run_id': self.course_id})+f'?message_count={message_count}',
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, 403)
+
+    @patch('learning_assistant.views.learning_assistant_enabled')
+    def test_learning_assistant_not_enabled(self, mock_learning_assistant_enabled):
+        mock_learning_assistant_enabled.return_value = False
+        message_count = 5
+        response = self.client.get(
+            reverse('message-history', kwargs={'course_run_id': self.course_id})+f'?message_count={message_count}',
+            content_type='application/json'
+        )
+
+        self.assertEqual(response.status_code, 403)
+
+    @patch('learning_assistant.views.learning_assistant_enabled')
+    @patch('learning_assistant.views.get_user_role')
+    @patch('learning_assistant.views.CourseEnrollment.get_enrollment')
+    @patch('learning_assistant.views.CourseMode')
+    def test_user_no_enrollment_not_staff(self, mock_mode, mock_enrollment, mock_role, mock_waffle):
+        mock_waffle.return_value = True
+        mock_role.return_value = 'student'
+        mock_mode.VERIFIED_MODES = ['verified']
+        mock_enrollment.return_value = None
+
+        message_count = 5
+        response = self.client.get(
+            reverse('message-history', kwargs={'course_run_id': self.course_id})+f'?message_count={message_count}',
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, 403)
+
+    @patch('learning_assistant.views.learning_assistant_enabled')
+    @patch('learning_assistant.views.get_user_role')
+    @patch('learning_assistant.views.CourseEnrollment.get_enrollment')
+    @patch('learning_assistant.views.CourseMode')
+    def test_user_audit_enrollment_not_staff(self, mock_mode, mock_enrollment, mock_role, mock_waffle):
+        mock_waffle.return_value = True
+        mock_role.return_value = 'student'
+        mock_mode.VERIFIED_MODES = ['verified']
+        mock_enrollment.return_value = MagicMock(mode='audit')
+
+        message_count = 5
+        response = self.client.get(
+            reverse('message-history', kwargs={'course_run_id': self.course_id})+f'?message_count={message_count}',
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, 403)
+
+    @patch('learning_assistant.views.learning_assistant_enabled')
+    @patch('learning_assistant.views.get_user_role')
+    @patch('learning_assistant.views.CourseEnrollment.get_enrollment')
+    @patch('learning_assistant.views.CourseMode')
+    @patch('learning_assistant.views.get_course_id')
+    def test_learning_message_history_view_get(self,
+                                               mock_get_course_id,
```

Comment:
NIT: This spacing looks a little strange to me, can you remove the indents?

Reply:
this was throwing a "line too long" quality error, I might just ignore it instead.

Reply:
Actually, I a better way by just moving the "self" param to the next line, then moving the spacing of all the params back to one tab past the function declaration.

Reply:
You can also keep them on the same line, they don't each need to be on their own!